src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+
+    private String getBaseUrl() {
+        return BASE_URL_PATH + port;
+    }
+
+    private String buildUrl(String... paths) {
+        return getBaseUrl() + "/" + String.join("/", paths);
+    }
+
+    private String getBaseResourceAvailabilityUrl() {
+        return buildUrl("resources", "availability");
+    }
+
+    private String getBaseResourceUrl() {
+        return "http://localhost:" + port + "/resources";
+    }
+
+    private String getBaseUserUrl() {
+        return "http://localhost:" + port + "/users";
+    }
+
+    private String getBaseProductUrl() {
+        return "http://localhost:" + port + "/products";
+    }
+
+    private String getProductUrl(UUID id) {
+        return getBaseUrl() + "/products/" + id;
+    }
+
+    @Autowired
+    TestRestTemplate testRestTemplate;
+    @Autowired
+    UserRepository userRepository;
+    @Autowired
+    ProductRepository productRepository;
+    @Autowired
+    ResourceRepository resourceRepository;
+    @Autowired
+    ResourceInUserRepository resourceInUserRepository;
+    @Autowired
+    ResourceInProductRepository resourceInProductRepository;
+
+    User user;
+    Gemstone gemstone;
+    ResourceInUserRequestDto resourceInUserRequestDto;
+    ResourcesInUserResponseDto resourcesInUser;
+    ProductRequestDto productRequestDto;
+    ProductResponseDto productResponseDto;
+
+    @BeforeEach
+    void setUp() {
+        userRepository.deleteAll();
+        productRepository.deleteAll();
+        resourceRepository.deleteAll();
+        resourceInUserRepository.deleteAll();
+        resourceInProductRepository.deleteAll();
+
+        user = getUser();
+        gemstone = getGemstone();
+        resourceInUserRequestDto = getResourceInUserRequestDto(user, Objects.requireNonNull(gemstone));
+        resourcesInUser = getResourcesInUserResponseDto(resourceInUserRequestDto);
+        productRequestDto = getProductRequestDto(Objects.requireNonNull(resourcesInUser), user);
+    }
+
+    @Test
+    void createProductSuccessfully() {
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.postForEntity(getBaseProductUrl(), productRequestDto, ProductResponseDto.class);
+        ProductResponseDto productResponseDto = response.getBody();
+
+        assertEquals(HttpStatus.CREATED, response.getStatusCode());
+
+        assertEquals(productRequestDto.getName(), Objects.requireNonNull(productResponseDto).getName());
+        assertEquals(productRequestDto.getResourcesContent().get(0).getId(), productResponseDto.getResourcesContent().get(0).getResource().getId());
+    }
+
+    @Test
+    void getProductSuccessfully() {
+
+        productResponseDto = getProductResponseDto(productRequestDto);
+
+        ResponseEntity<ProductResponseDto> response =
+                this.testRestTemplate.getForEntity(getProductUrl(Objects.requireNonNull(productResponseDto).getId()), ProductResponseDto.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
```

Comment:
We should add more assertions for the result of a test. We should somehow compare the object that's coming back from the response with what we expect.

Reply:
Done!